### PR TITLE
Request for comment

### PR DESCRIPTION
**Pull request by mosen**
Hi Graham,
In this version i removed the need for jQuery by using the Fetch API which has been part of Chrome since v42.
I also used a different approach to Asynchronous methods, removing the semaphore based method you had and replacing all of those with chained Promises.

Let me know what you think, there is no need to merge as not everything is complete.

### CODE REVIEW

**Comment by grahamgilbert** [comment]
From a quick scroll through on my phone, I like this style much better.

**Review by grahamgilbert — CHANGES_REQUESTED** [review]
* Not sure why a settings page has been added. This gives the illusion that users can manage their own settings. I cannot see a situation where I would want users to configure this themselves.
* The inventory URL isn't configurable on the Sal server, so not sure why it would be configurable here?

Tldr: I don't think local settings are useful beyond the settings.json for local development.

**Comment by mosen** [comment]
Hi graham. I have also selfish reasons for sending the PR. that is, to merge my own extension into chromesal so that there is no duplication of effort. Part of that was a settings page I had to test different servers. As you said, this is probably against your design for sal itself.

**Review by grahamgilbert — APPROVED** [review]

**Comment by grahamgilbert** [comment]
So this is looking pretty awesome now. Let me know when you're done and I'll start reviewing.

**Comment by mosen** [comment]
Options page was wound back but still need to test it against live Sal

**Comment by grahamgilbert** [comment]
How are you doing with this mr @mosen?

**Comment by grahamgilbert** [comment]
Pinging again mr @mosen
